src/connection.cpp

```diff
@@ -750,13 +856,47 @@ void Connection::send_initial_auth_response() {
   }
 }
 
-void Connection::PendingSchemaAgreement::stop_timer() {
-  if (timer != NULL) {
-    Timer::stop(timer);
-    timer = NULL;
+void Connection::restart_heartbeat_timer() {
+  if (config_.connection_heartbeat_interval_secs() > 0) {
+    heartbeat_timer_.start(loop_,
+                           1000 * config_.connection_heartbeat_interval_secs(),
+                           this, on_heartbeat);
   }
 }
 
+void Connection::on_heartbeat(Timer* timer) {
+  Connection* connection = static_cast<Connection*>(timer->data());
+
+  if (connection->idle_time_ms_ == 0) {
```

Comment:
I originally read this as "amount of idle time". Would it be appropriate to call it ``idle_start_time_`` to help dummies like me understand that it is absolute time?